src/mock.ts

```diff
@@ -0,0 +1,442 @@
+  ) => {
+    callback?.(true);
+    return toValue;
+  },
+  withTiming: (
+    toValue: AnimatableValue,
+    _userConfig?: WithTimingConfig,
+    callback?: AnimationCallback
+  ) => {
+    callback?.(true);
+    return toValue;
+  },
+};
+
+const interpolation = {
+  Extrapolation,
+  interpolate: NOOP,
+  clamp: NOOP,
+};
+
+const interpolateColor = {
+  Extrapolate: Extrapolation,
+  Extrapolation,
+  ColorSpace,
+  interpolateColor: NOOP,
+  // useInterpolateConfig: ADD ME IF NEEDED
+};
+
+const Easing = {
+  Easing: {
+    linear: ID,
+    ease: ID,
+    quad: ID,
+    cubic: ID,
+    poly: ID,
+    sin: ID,
+    circle: ID,
+    exp: ID,
+    elastic: ID,
+    back: ID,
+    bounce: ID,
+    bezier: () => ({ factory: ID }),
+    bezierFn: ID,
+    steps: ID,
+    in: ID,
+    out: ID,
+    inOut: ID,
+  },
+};
+
+const platformFunctions = {
+  measure: () => ({
+    x: 0,
+    y: 0,
+    width: 0,
+    height: 0,
+    pageX: 0,
+    pageY: 0,
+  }),
+  // dispatchCommand: ADD ME IF NEEDED
+  // scrollTo: ADD ME IF NEEDED
+  // setGestureState: ADD ME IF NEEDED
+  // setNativeProps: ADD ME IF NEEDED
+  // getRelativeCoords: ADD ME IF NEEDED
+};
+
+const Colors = {
+  // isColor: ADD ME IF NEEDED
+  processColor: processColorRN,
+  // convertToRGBA: ADD ME IF NEEDED
+};
+
+const PropAdapters = {
+  // createAnimatedPropAdapter: ADD ME IF NEEDED
+};
+
+class BaseAnimationMock {
+  duration() {
+    return this;
+  }
+
+  delay() {
+    return this;
+  }
+
+  springify() {
+    return this;
+  }
+
+  damping() {
+    return this;
+  }
+
+  stiffness() {
+    return this;
+  }
+
+  withCallback() {
+    return this;
+  }
+
+  randomDelay() {
+    return this;
+  }
+
+  withInitialValues() {
+    return this;
+  }
+
+  easing(_: (t: number) => number) {
+    return this;
+  }
+
+  rotate(_: string) {
+    return this;
+  }
+
+  mass(_: number) {
+    return this;
+  }
+
+  restDisplacementThreshold(_: number) {
+    return this;
+  }
+
+  restSpeedThreshold(_: number) {
+    return this;
+  }
+
+  overshootClamping(_: number) {
+    return this;
+  }
+
+  dampingRatio(_: number) {
+    return this;
+  }
+
+  getDelay() {
+    return 0;
+  }
+
+  getDelayFunction() {
+    return NOOP;
+  }
+
+  getDuration() {
+    return 300;
+  }
+
+  getReduceMotion() {
+    return ReduceMotion.System;
+  }
+
+  getAnimationAndConfig() {
+    return [NOOP, {}];
+  }
+
+  build() {
+    return () => ({ initialValues: {}, animations: {} });
+  }
+
+  reduceMotion() {
+    return this;
+  }
+}
+
+const core = {
+  runOnJS: ID,
+  runOnUI: ID,
+  createWorkletRuntime: NOOP,
+  runOnRuntime: NOOP,
+  makeMutable: ID,
+  makeShareableCloneRecursive: ID,
+  isReanimated3: () => true,
+  // isConfigured: ADD ME IF NEEDED
+  enableLayoutAnimations: NOOP,
+  // getViewProp: ADD ME IF NEEDED
+};
+
+const layoutReanimation = {
+  BaseAnimationBuilder: new BaseAnimationMock(),
+  ComplexAnimationBuilder: new BaseAnimationMock(),
+  Keyframe: new BaseAnimationMock(),
```

Comment:
Why is this preferred over our previous solution
```tsx
].forEach((k) =>
  Object.assign(ReanimatedV2, {
    [k]: new BaseAnimationMock(),
  })
);
```
?
Anyway, you can specify type, to make sure nobody makes a typo when assigning  baseAnimations:

```suggestion
const layoutReanimation: Record<string, BaseAnimationMock> = {
  BaseAnimationBuilder: new BaseAnimationMock(),
  ComplexAnimationBuilder: new BaseAnimationMock(),
  Keyframe: new BaseAnimationMock(),
```